pulser-core/pulser/sequence/_schedule.py

```diff
@@ -261,7 +261,9 @@ def add_pulse(
         t0 = last.tf
         current_max_t = max(t0, *phase_barrier_ts)
         for ch, ch_schedule in self.items():
-            if protocol == "no-delay" or ch == channel:
+            if protocol == "no-delay":
+                break
```

Comment:
This could be put outside of the for-loop in this way:
```python
if protocol != "no-delay":
    ... # code of the for-loop
```
Or, even encompassing the subsequent lines:
```python
phase_jump_buffer = 0
if protocol != "no-delay":
    ... # code of the for-loop
    ... # code of the try except
```
(Ideally, the for-loop would also be put in a separate function)

Reply:
Yes, I tried that but it pushed the code too much to the right...

Reply:
Ok, I followed your suggestion with the changes in the (hopefully) last commit. Wdyt @CdeTerra ?